src/writer/autogen.py

```
@@ -253,19 +335,200 @@ def _get_main_prompt(description: str):
+            "role": "user",
+            "content": "A generate_blueprint tool call is required.",
+        })
+        return None
+
+    call_args = _decode_tool_args(tool_call[0], autogen_state)
+    if call_args is None:
+        return None
+
+    try:
+        id_to_generated_node = _validate_blueprint_nodes(call_args.get("components"))
+    except BaseException as exception:
```

Pylint (3.3.7)

[warning] Catching too general exception BaseException

(W0718)